src/Sylius/Behat/Context/Api/Admin/ManagingPromotionsContext.php

```diff
@@ -178,4 +525,300 @@ public function thePromotionShouldNotAppliesToDiscountedItems(PromotionInterface
+                'channels',
+                $this->sectionAwareIriConverter->getIriFromResourceInSection($channel, 'admin'),
+            ),
+            sprintf('Catalog promotion is not assigned to %s channel', $channel->getName()),
+        );
+    }
+
+    /**
+     * @When the :promotion promotion should have a label :label in :localeCode locale
+     */
+    public function thePromotionShouldHaveLabelInLocale(PromotionInterface $promotion, string $label, string $localeCode): void
+    {
+        $response = $this->client->show(Resources::PROMOTIONS, $promotion->getCode());
+
+        Assert::true($this->responseChecker->hasTranslation($response, $localeCode, 'label', $label));
+    }
+
+    /**
+     * @Then /^it should have ("[^"]+") of item percentage discount configured for ("[^"]+" channel)$/
+     */
+    public function itShouldHaveOfItemPercentageDiscount(float $percentage, ChannelInterface $channel): void
+    {
+        $actions = $this->responseChecker->getValue($this->client->getLastResponse(), 'actions');
+        foreach ($actions as $action) {
+            if ($action['type'] === 'unit_percentage_discount') {
+                Assert::same((float) $action['configuration'][$channel->getCode()]['percentage'], $percentage);
+            }
+        }
+    }
+
+    /**
+     * @Then /^it should have ("[^"]+") of order percentage discount$/
+     */
+    public function itShouldHaveOfOrderPercentageDiscount(float $percentage): void
+    {
+        $actions = $this->responseChecker->getValue($this->client->getLastResponse(), 'actions');
+        Assert::same((float) $actions[0]['configuration']['percentage'], $percentage);
+    }
+
+    /**
+     * @Then the code field should be disabled
+     */
+    public function theCodeFieldShouldBeDisabled(): void
+    {
+        $this->client->updateRequestData(['code' => 'NEW_CODE']);
+
+        Assert::false($this->responseChecker->hasValue($this->client->update(), 'code', 'NEW_CODE'));
+    }
+
+    /**
+     * @Then the :promotion promotion should be available from :startsDate to :endsDate
+     */
+    public function thePromotionShouldBeAvailableFromTo(
+        PromotionInterface $promotion,
+        \DateTimeInterface $startsDate,
+        \DateTimeInterface $endsDate
+    ): void {
+        Assert::true(
+            $this->responseChecker->hasItemWithValues(
+                $this->client->index(Resources::PROMOTIONS),
+                [
+                    'name' => $promotion->getName(),
+                    'startsAt' => $startsDate->format('Y-m-d H:i:s'),
+                    'endsAt' => $endsDate->format('Y-m-d H:i:s'),
+                ],
+            ),
+            sprintf(
+                'Cannot find promotions with name "%s" available between "%s" and "%s" in the list',
+                $promotion->getName(),
+                $startsDate->format('Y-m-d H:i:s'),
+                $endsDate->format('Y-m-d H:i:s'),
+            ),
```

Comment:
This may be an unpopular opinion 😅 , but in cases where we are only checking one thing and the name of the step describes exactly what is being checked, I think this additional message is completely unnecessary. If a step does not pass, we will know exactly what happened and this comment does not clarify anything.